extensions/amp-skimlinks/0.1/link-rewriter/link-rewriter-manager.js

```diff
@@ -0,0 +1,284 @@
+   * @param {?{linkSelector: string}=} options
+   *   - linkSelector is an optional CSS selector to restrict
+   *    which anchors the link rewriter should handle.
+   *    Anchors not matching the CSS selector will be ignored.
+   *    If not provided the link rewrite will handle all the links
+   *    found on the page.
+   *
+   * @return {!./link-rewriter.LinkRewriter}
+   * @public
+   */
+  registerLinkRewriter(linkRewriterId, resolveUnknownLinks, options) {
+    const linkRewriter = new LinkRewriter(
+        this.ampPageDocument_,
+        linkRewriterId,
+        resolveUnknownLinks,
+        options
+    );
+    this.insertInListBasedOnPriority_(this.linkRewriters_, linkRewriter,
+        this.priorityList_);
+    // Trigger initial scan.
+    linkRewriter.onDomUpdated();
+
+    return linkRewriter;
+  }
+
+
+  /**
+   * Notify the LinkRewriteService that a click has happened on an anchor.
+   * This is the chance for one of the registered LinkRewriter to rewrite the
+   * url before the browser handles the click and navigates to the link url.
+   *
+   * The LinkRewriter allowed to replace the url will be chosen based on the
+   * defined priority order (in link attribute or global meta tag).
+   * If a LinkRewriter decides to not replace the url,  we will ask the next
+   * one if it wants to replace the link until one replacement is made.
+   *
+   * A "CLICK" event will also be dispatched to notify listeners that a click
+   * on an anchor has happened. This should mostly be used to send click
+   * tracking requests, handlers of this events should not
+   * mutate the anchor!
+   * @param {!HTMLElement} anchor
+   * @param {string} clickType - 'click' or 'contextmenu'
+   * @public
+   */
+  maybeRewriteLink(anchor, clickType = 'click') {
+    const suitableLinkRewriters = this.getSuitableLinkRewritersForLink_(anchor);
+    if (suitableLinkRewriters.length) {
+      let chosenLinkRewriter = null;
+
+      // Iterate by order of priority until one of the linkRewriter
+      // replaces the link.
+      for (let i = 0; i < suitableLinkRewriters.length; i++) {
+        const hasReplaced = suitableLinkRewriters[i].rewriteAnchorUrl(anchor);
+        if (hasReplaced) {
+          chosenLinkRewriter = suitableLinkRewriters[i];
+          break;
+        }
+      }
+      const linkRewriterId = chosenLinkRewriter ? chosenLinkRewriter.id : null;
+
+      // Emit click event for analytics purposes only,
+      // anchor should not me mutated by handlers.
+      const eventData = {
+        linkRewriterId,
+        anchor,
+        clickType,
+      };
+
+      suitableLinkRewriters.forEach(linkRewriter => {
+        const event = {
+          type: EVENTS.CLICK,
+          eventData,
+        };
+
+        linkRewriter.events.fire(event);
+      });
+    }
+  }
+
+  /**
+   * Extract the priority list from the optional html meta tag.
+   * The meta tag should contain a whitespace separated list of
+   * LinkRewriter Ids.
```

Comment:
is the usage of this meta tag documented somewhere? can we add it to examples/amp-skimlinks.html

Reply:
This meta tag is not mandatory and will not be useful unless the link-rewriter solution is made available to more extensions  (which  is not currently the case since I've been asked to bring the code back to `amp-skimlinks`).

I'll add the meta tag to the `amp-skimlinks.html` as an example though but we should have a separate discussion (later) about if you are interested in keeping the generic part of the link-rewriter. If not the meta-tag will go away along with everything related to the priority list.